main.go

```diff
@@ -70,40 +73,20 @@ func main() {
 	}
 	id := strings.TrimSpace(fmt.Sprint(cmd.Stdout)) // docker image id
 
-	// Run protoc (in a container) on each .proto file.
-	//
-	// The explicit PWDs are required to appease protoc's
-	// rather sensitive file name expectations.
-	//
-	// All files in a single protoc invocation must belong
-	// to the same proto package, hence the loop.
-	found := false
-	filepath.Walk("proto", func(path string, info fs.FileInfo, err error) error {
-		if err != nil {
-			return err
-		}
-		if strings.HasSuffix(info.Name(), ".proto") {
-			log.Printf("compiling %s...", path)
-			// We assume pwd does not conflict with some critical part
-			// of the docker image, and volume-mount it.
-			found = true
-			cmd := exec.Command("docker", "run", "-v", pwd+":"+pwd, id,
-				"--proto_path="+pwd+"/proto",
-				"--go_out="+pwd,
-				"--twirp_out="+pwd,
-				"--go_opt=paths=source_relative",
-				pwd+"/"+path,
-			)
-			cmd.Stderr = os.Stderr
-			cmd.Stdout = os.Stderr
-			if err := cmd.Run(); err != nil {
-				log.Fatalf("protoc command (%s) failed: $v", err)
-			}
-		}
-		return nil
-	})
-	if !found {
-		log.Fatal("found no .proto files")
+	// Log the command, neatly.
+	protocArgs := flag.Args()
+	cmdstr := "protoc " + strings.ReplaceAll(strings.Join(protocArgs, " "), pwd, "$(pwd)")
+	log.Println(cmdstr)
+
+	// Run protoc, in a container.
+	// We assume pwd does not conflict with some critical part
+	// of the docker image, and volume-mount it.
+	cmd = exec.Command("docker", "run", "-v", pwd+":"+pwd, id)
```

Comment:
This isn't related, but while looking at this I was struck by the idea of embedding the Dockerfile instead of including it inline. I did so in #4. What do you think?